test/javascripts/page_test.coffee

```diff
@@ -85,3 +85,110 @@ describe 'Page', ->
         onlyKeys: ['a']
 
       @replaceStateStub.calledWith mockXHR.getResponseHeader('X-XHR-Redirected-To')
+
+  describe 'onReplace', ->
+
+    beforeEach ->
+      @fakebody = """
+        <div id='fakebody'>
+          <div id='foo'>
+            <div id='bar'>
+              <div id='foobar'></div>
+            </div>
+          </div>
+          <div id='baz'>
+            <div id='bat'></div>
+          </div>
+        </div>
+      """
+      $('body').append(@fakebody)
+
+    afterEach ->
+      $('#fakebody').remove()
+
+
+    it 'calls the onReplace function only once', ->
+      node = $('#foo')[0]
+      refreshing = [$("#fakebody")[0]]
+      callback = stub()
+
+      Page.onReplace(node, callback)
+
+      triggerEvent 'page:before-partial-replace', refreshing
+      triggerEvent 'page:before-replace'
+
+      assert callback.calledOnce
+
+    it 'calls the onReplace function only once, even if someone were to mess with the number of events fired', ->
+      node = $('#foo')[0]
+      refreshing = [$("#fakebody")[0]]
+      callback = stub()
+
+      Page.onReplace(node, callback)
+
+      triggerEvent 'page:before-partial-replace', refreshing
+      triggerEvent 'page:before-partial-replace', refreshing
+      triggerEvent 'page:before-partial-replace', refreshing
+      triggerEvent 'page:before-replace'
+      triggerEvent 'page:before-replace'
+      triggerEvent 'page:before-replace'
+
+      assert callback.calledOnce
+
+    it 'throws an error not supplied enough arguments', ->
+      try
```

Comment:
I think it's a lot more straightforward to test the errors this way (test em as you'd use em)

Is that alright?  Not sure that I could test the string message without a lot of syntax